src/python/dxpy/bindings/apollo/ParseFiltersJSONbySchema.py

```diff
@@ -0,0 +1,506 @@
+
+    Filters must be defined in schema["filtering_conditions"]
+
+    There are currently three ways to define filtering_conditions when version is 1.0:
+    1. Basic use-case: no "properties" are defined. Only "condition", "table_column" and optionally "max_item_limit" are defined.
+                       In this case, there are no sub-keys for the current key in input_json.
+                       (see 'sample_id' in 'assay_filtering_conditions.EXTRACT_ASSAY_EXPRESSION_FILTERING_CONDITIONS')
+    2. Properties defined as dict of dicts (see 'annotation' and 'expression' in EXTRACT_ASSAY_EXPRESSION_FILTERING_CONDITIONS)
+        2.1. If filters_combination_operator is not defined, the assumption is that there is only one sub-key in input_json
+
+    3. Complex use-case: Properties defined as list of dicts (more advanced, special use-case with complex conditional logics that needs translation)
+        Filtering conditions defined this way indicate that input_json may contain more than one item for the current key, and elements are stored in a list.
+        In this case, the schema must define "items_combination_operator" and "filters_combination_operator".
+        items_combination_operator: how to combine filters for list items in input_json
+        filters_combination_operator: how to combine filters within each item
+        (see 'location' in EXTRACT_ASSAY_EXPRESSION_FILTERING_CONDITIONS)
+
+        Within 'properties' if "key" is defined, then a generic one-key filter is built.
+        If "keys" is defined, then more complex use-cases are handled via special conditions that are defined in condition_function_mapping.
+
+    """
+
+    def __init__(self, input_json, schema, error_handler=print):
+        self.input_json = input_json
+        self.schema = schema
+        self.error_handler = error_handler
+        self.condition_function_mapping = {
+            "genobin_partial_overlap": self.build_partial_overlap_genobin_filters,
+        }
+        self.SUPPORTED_VIZSERVER_CONDITIONS = [
+            "contains",
+            "exists",
+            "not-exists",
+            "any",
+            "not-any",
+            "all",
+            "not-empty",
+            "in",
+            "not-in",
+            "is",
+            "is-not",
+            "greater-than",
+            "less-than",
+            "greater-than-eq",
+            "less-than-eq",
+            "between",
+            "between-ex",
+            "between-left-inc",
+            "between-right-inc",
+            "not-between",
+            "not-between-ex",
+            "not-between-left-inc",
+            "not-between-right-inc",
+            "compare-before",
+            "compare-after",
+            "compare-within",
+        ]
+
+    def parse(self):
+        self.is_valid_json(self.schema)
+        if self.get_schema_version(self.schema) == "1.0":
+            return self.parse_v1()
+        else:
+            raise NotImplementedError
+
+    def parse_v1(self):
```

Comment:
Minor: A top-level `try .. except` might help to wrap expected errors into `ParsingError` to still have controlled way how to end unsuccessful parsing attempt.